src/triton_fil/model_instance_state.cuh

```diff
@@ -58,6 +59,29 @@ class ModelInstanceState : public BackendModelInstance {
       TRITONBACKEND_ModelInstance* triton_model_instance, const char* name,
       const int32_t device_id);
 
+  template <class Iter>
+  typename std::enable_if_t<std::is_same<
```

Comment:
I don't understand why SFINAE is needed here. Since you are essentially doing `std::is_same` for exactly one type, can't that type just be the function overload?

Reply:
The `enable_if_t` is checking if the value inside the iterator is a `TRITONBACKEND_Request*`, but we want to enable this for all containers that contain such objects. If we had the function just accept an iterator of a specific type, wouldn't we have to be specific about which container's iterator we're working with?

To be honest, I didn't think about this too much except that this is the way I've generally implemented templates that don't care about the details of the iterator type but do care about the contents of that iterator in the past. Is there a better way to do this and/or did I overthink it here?

Reply:
I misread the intention of the SFINAE here, I think the pointer type threw me off here. It's an iterator container, whose each element is a pointer to a `TRITORBACKEND_Request`. That makes much more sense, thanks!